bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -250,6 +250,47 @@ impl Room {
         }
     }
 
+    pub fn send_read_receipt(&self, event_id: String) -> Result<()> {
+        let room = match &self.room {
+            SdkRoom::Joined(j) => j.clone(),
+            _ => bail!("Can't send read receipts to room that isn't in joined state"),
+        };
+
+        let event_id = EventId::parse(event_id)?;
+
+        RUNTIME.block_on(async move {
+            room.read_receipt(&event_id).await?;
+            Ok(())
+        })
+    }
+
+    pub fn send_read_marker(
+        &self,
+        fully_read_event_id: String,
+        read_receipt_event_id: Option<String>,
+    ) -> Result<()> {
+        let room = match &self.room {
+            SdkRoom::Joined(j) => j.clone(),
+            _ => bail!("Can't send read markers to room that isn't in joined state"),
+        };
+
+        let fully_read = EventId::parse(fully_read_event_id)?;
+        let read_receipt: Option<OwnedEventId>;
+        if let Some(e) = read_receipt_event_id {
+            match EventId::parse(e) {
+                Ok(e) => read_receipt = Some(e),
+                Err(_) => bail!("Invalid event ID for read receipt."),
+            };
+        } else {
+            read_receipt = None
+        }
```

Comment:
If you don't care to have a different error msg for invalid `fully_read_event_id` or invalid `read_receipt_event_id`, you can do

```suggestion
        let read_receipt = read_receipt_event_id.map(EventId::parse).transpose()?;
```

If you want separate error messages, I would recommend adding `.context("parsing read receipt event ID")` before the `?` and another `.context` before the other `?`

Reply:
Thanks, that's *much* neater, `transpose` looks useful 👍